### PR TITLE
Implement multisite theming and settings

### PR DESCRIPTION
**Pull request by tuchfarber**
App now retrieves theming, user, and site information from server and dynamically renders the page with that information. Current theming work only works with production builds.

Relies on edx/journals#100

### CODE REVIEW

File: src/components/Header/index.jsx [hunk 1]
```diff
@@ -1,26 +1,37 @@
...
 
 import './Header.scss';
 
 class Header extends React.Component {
   componentDidMount() {
-    this.props.getUserInfo();
+    this.props.getSiteInfo();
   }
   render() {
+    const stylesheet = `${this.props.themeName}-app.css`;
```
Comment (bill-filler) [inline src/components/Header/index.jsx]: Just so I understand: So we are fetching the theme name from the server, but the actual theme file will be part of the build of the frontend app?

Comment (tuchfarber) [reply src/components/Header/index.jsx]: Correct, theme name will be part of the SiteBranding on the server, but <theme>-app.css will be exported with the rest of the files in the frontend app's prod build.

File: src/components/Header/index.jsx [hunk 2]
```diff
@@ -1,26 +1,37 @@
...
-          <Link to="/">Home</Link>
-        </h3>
-        {
-          this.props.isAuthenticated ? (
-            <span>{this.props.username}</span>
-          ) : (
-            <a href={this.props.loginPath}>Login</a>
-          )
-        }
+        <Helmet>
```
Comment (bill-filler) [inline src/components/Header/index.jsx]: nice

File: src/themes/README.md [hunk 3]
```diff
@@ -1,5 +1,34 @@
...
+Inside of this file, you can simply override variables `$primary = #ff0000`, and anywhere that Bootstrap using that primary color, the new color will be used. Any new variable added in a theme should also have a variable in the base.scss with `!default` so it has a fallback. Once the theme is ready, run the production build (see section below) and `/dist/` will now contain a `base-app.css` and `{themename}-app.css`. These are complete themes in the sense that they contain all necessary CSS for the site.
+
+### Production builds
+Due to how webpack is handling development build currently (hot reloading), theming is only available in the production build. In order to use the production build, you need to do the following:
+  1.
+  1. Allow a new site in the CORS_ORIGIN_WHITELIST setting of the backend: `CORS_ORIGIN_WHITELIST = ('localhost:1991','localhost:8000')`
+  2. Restart backend: `docker-compose journals restart`
+  3. Enter frontend container: `make shell`
+  4. Set the a JOURNALS_BASE_URL enviornment variable equal to the development backend host: `JOURNALS_BASE_URL='http://localhost:18606'`
+  5. Run production build `npm run build`
```
Comment (bill-filler) [inline src/themes/README.md]: need:
```
npm install simplehttpserver
```

Comment (bill-filler) [reply src/themes/README.md]: actually this doesn't work, it needs it's deps as well

File: src/themes/README.md [hunk 4]
```diff
@@ -1,5 +1,34 @@
...
+  6. Host the files in the `dist` directory: `cd dist; python -m SimpleHttpServer` (or similar)
```
Comment (bill-filler) [inline src/themes/README.md]: add step before this to exit the container

File: src/themes/README.md (hunk 3 above)
Comment (bill-filler) [reply src/themes/README.md]: ignore

**Comment by bill-filler** [comment]
working now after following readme. Still seeing one error when loading initial page, but not sure how important it is:
```
 action STARTED_FETCHING_SITE_INFO @ 17:12:55.256
redux-logger.js:1  action STARTED_FETCHING_JOURNALS @ 17:12:55.263
HelmetUtils.js:380 GET http://localhost:8000/-app.css 404 (File not found)
(anonymous) @ HelmetUtils.js:380
_ @ HelmetUtils.js:379
x @ HelmetUtils.js:252
(anonymous) @ HelmetUtils.js:222
requestAnimationFrame (async)
t.handleClientStateChange @ HelmetUtils.js:221
f @ index.js:45
t.componentWillMount @ index.js:85
mountClassInstance @ react-dom.production.min.js:110
beginWork @ react-dom.production.min.js:133
r @ react-dom.production.min.js:161
a @ react-dom.production.min.js:161
i @ react-dom.production.min.js:162
S @ react-dom.production.min.js:169
w @ react-dom.production.min.js:168
h @ react-dom.production.min.js:167
d @ react-dom.production.min.js:165
updateContainer @ react-dom.production.min.js:175
(anonymous) @ react-dom.production.min.js:225
unbatchedUpdates @ react-dom.production.min.js:171
Ko @ react-dom.production.min.js:225
render @ react-dom.production.min.js:227
194 @ index.jsx:35
f @ bootstrap:78
193 @ journal.js:20
f @ bootstrap:78
t @ bootstrap:45
r @ bootstrap:32
(anonymous) @ app.min.js?c8fa28b8281d82f6e195:1
redux-logger.js:1  action GET_SITE_INFO_SUCCESS @ 17:12:55.369
redux-logger.js:1  action FINISHED_FETCHING_SITE_INFO @ 17:12:55.375
redux-logger.js:1  action GET_JOURNALS_SUCCESS @ 17:12:55.391
redux-logger.js:1  action FINISHED_FETCHING_JOURNALS @ 17:12:55.395
```

**Pull request merged**